In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

**Se crea el dataframe data_diabetes.**

In [4]:
data_diabetes = pd.read_csv("./data/Diabetes_es_ext.csv")


**Muestra la lista de columnas del dataset data_diabetes.**

In [5]:
list(data_diabetes)

['Embarazos',
 'Glucosa',
 'Presión sanguínea',
 'Pliegue cutáneo',
 'Insulina',
 'Índice de masa corporal',
 'Pedigrí diabetes',
 'Edad',
 'Diabetes',
 'Medicación previa',
 'Observaciones',
 'Fecha de diagnóstico']

**Muestra la descripción de las columnas del dataset con sus respectivos datos estadísticos. Trasponemos los datos para mejor visualización.**

In [6]:
data_diabetes.describe().T

count        mean         std   min   25%    50%     75%  \
Embarazos          768.0    3.845052    3.369578   0.0   1.0    3.0    6.00   
Glucosa            768.0  120.894531   31.972618   0.0  99.0  117.0  140.25   
Presión sanguínea  768.0   69.105469   19.355807   0.0  62.0   72.0   80.00   
Pliegue cutáneo    768.0   20.536458   15.952218   0.0   0.0   23.0   32.00   
Insulina           768.0   79.799479  115.244002   0.0   0.0   30.5  127.25   
Edad               768.0   33.240885   11.760232  21.0  24.0   29.0   41.00   

                     max  
Embarazos           17.0  
Glucosa            199.0  
Presión sanguínea  122.0  
Pliegue cutáneo     99.0  
Insulina           846.0  
Edad                81.0

**Se filtran pacientes con edades mayores a 20 y menores e iguales de 45 años y que hayan sido diagnosticados con diabetes. Además se mostrarán solo las columnas: Embarazos, Edad,	Glucosa, Insulina y Presión sanguínea.**

In [7]:
df_d = data_diabetes.query('Edad > 20 and Edad <= 45 and Diabetes =="Sí"')[["Embarazos","Edad","Glucosa","Insulina", "Presión sanguínea"]]

***Se muestra la lista de datos del query anterior,ordenados por la columna Edad.***

In [8]:
df_d.sort_values(["Edad"])

Embarazos  Edad  Glucosa  Insulina  Presión sanguínea
580          0    21      151         0                 90
414          0    21      138       167                 60
220          0    21      177       478                 60
577          2    21      118         0                 80
255          1    21      113         0                 64
..         ...   ...      ...       ...                ...
743          9    45      140         0                 94
387          8    45      105         0                100
708          9    45      164         0                 78
493          4    45      125       122                 70
590         11    45      111         0                 84

[210 rows x 5 columns]

**Con la funcion .info(), se muestran las columnas del dataset, con sus respectivos tipos de datos. Tenemos 12 campos en total, 6 campos de tipo entero (int64) y 6 de tipo objeto (object). El rango de los índices van de 0 a 767 (RangeIndex: 768 entries, 0 to 767) lo cual me indica que contiene 768 filas. Con un uso de memoria de 72.1+ KB (memory usage: 72.1+ KB).**

In [9]:
data_diabetes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Embarazos                768 non-null    int64 
 1   Glucosa                  768 non-null    int64 
 2   Presión sanguínea        768 non-null    int64 
 3   Pliegue cutáneo          768 non-null    int64 
 4   Insulina                 768 non-null    int64 
 5   Índice de masa corporal  768 non-null    object
 6   Pedigrí diabetes         768 non-null    object
 7   Edad                     768 non-null    int64 
 8   Diabetes                 768 non-null    object
 9   Medicación previa        14 non-null     object
 10  Observaciones            13 non-null     object
 11  Fecha de diagnóstico     768 non-null    object
dtypes: int64(6), object(6)
memory usage: 72.1+ KB


***La propiedad .shape muestra las dimensiones del dataset, para el ejemplo tenemos que data_diabetes cuenta con 768 filas y 12 columnas es decir una matriz de 768x12.***

In [10]:
data_diabetes.shape

(768, 12)

In [11]:
data_diabetes.query('Glucosa >=78 and Glucosa <=85 and Diabetes =="Sí"').count()

Embarazos                  4
Glucosa                    4
Presión sanguínea          4
Pliegue cutáneo            4
Insulina                   4
Índice de masa corporal    4
Pedigrí diabetes           4
Edad                       4
Diabetes                   4
Medicación previa          0
Observaciones              0
Fecha de diagnóstico       4
dtype: int64